mysql-test/suite/rocksdb/t/type_char_secondary_index_only_scan.test

```diff
@@ -0,0 +1,50 @@
+# 1. Find all collations that support index-only scans with secondary
+#    keys using the "Extra" field from "EXPLAIN SELECT"
+# 2. Verify if the original string is reconstructed correctly from SK
+#    (or from PK when index-only scans with SK are not supported)
+
+--source include/have_rocksdb.inc
+
+CREATE TABLE collations AS SELECT * FROM information_schema.COLLATION_CHARACTER_SET_APPLICABILITY WHERE COLLATION_NAME NOT IN ("latin2_czech_cs", "cp1250_czech_cs") ORDER BY COLLATION_NAME;
+
+--disable_query_log
+--let $i=1
+--let $charset = query_get_value(select * from collations, CHARACTER_SET_NAME, $i)
+--let $collate = query_get_value(select * from collations, COLLATION_NAME, $i)
+--echo CharSet, Collate, Char Extra, VarChar Extra
+while ($charset != 'No such row')
+{
+  --let table_name=t_$charset
+  if ($charset == "binary")
+  {
+    --eval CREATE TABLE $table_name (id INT NOT NULL AUTO_INCREMENT PRIMARY KEY, vch VARBINARY(64), ch BINARY(64), KEY vch_idx (vch), KEY ch_idx (ch)) ENGINE=rocksdb
+  }
+  if ($charset != "binary")
+  {
+    --eval CREATE TABLE $table_name (id INT NOT NULL AUTO_INCREMENT PRIMARY KEY, vch VARCHAR(64), ch CHAR(64), KEY vch_idx (vch), KEY ch_idx (ch)) ENGINE=rocksdb CHARSET $charset COLLATE $collate
+  }
+  --eval INSERT INTO $table_name (vch, ch) VALUES('VARCHAR', 'char')
+  --let $vch_extra = query_get_value(EXPLAIN SELECT vch FROM $table_name, Extra, 1)
```

Comment:
$table_name has a single record and we are checking if the optimizer has picked index or not. This seems very flaky test case as we can't bet on optimizer to always pick index for signal record.

Reply:
This test tries to verify if data is reconstructed correctly from:
1. SK using `unpack_info` or mem-comparable form (`Using index`)
2. PK when index-only scans with SK are not supported (`EXPLAIN SELECT` returns `NULL`).
The server is deterministic so it should always return the same results unless optimizer will be changed in the future, but I assume we would like to know about it.